react/components/ModalDialog/index.test.tsx

```diff
@@ -0,0 +1,166 @@
+import React from 'react'
+import { render, fireEvent } from '@testing-library/react'
+
+import ModalDialog from '.'
+
+describe('ModalDialog', () => {
+  jest.spyOn(window, 'scroll').mockImplementation()
+  it('should have a default export', () => {
+    expect(typeof ModalDialog).toBe('function')
+  })
+
+  describe('cancelation', () => {
+    it('should render label', () => {
+      const containerModal = document.createElement('div')
+      const onCancel = jest.fn()
+      const cancelLabel = 'cancel'
+      const { getByText, container } = render(
+        <ModalDialog
+          isOpen
+          container={containerModal}
+          cancelation={{
+            label: cancelLabel,
+            onClick: onCancel,
+          }}
+          confirmation={{
+            label: 'confirm',
+            onClick: () => null,
+          }}
+        >
+          Foo
+        </ModalDialog>,
+        { container: containerModal }
+      )
+      expect(getByText(cancelLabel)).not.toBeNull()
+      expect(container).toMatchSnapshot()
+    })
+
+    it('onClick should be called on cancelation button click', () => {
+      const onCancel = jest.fn()
+
+      const { getByText } = render(
+        <ModalDialog
+          isOpen
+          cancelation={{
+            label: 'cancel',
+            onClick: onCancel,
+          }}
+          confirmation={{
+            label: 'confirm',
+            onClick: () => null,
+          }}
+        >
+          Foo
+        </ModalDialog>,
+        { container: document.body }
```

Comment:
I couldn't remove the`document.body` here because without it I can't click the button, but I believe that when I rewrite the component I'll be able to capture without using this